Source/Core/Core/IOS/Network/KD/Mail/WC24Send.cpp

```diff
@@ -61,6 +64,118 @@ bool WC24SendList::CheckSendList() const
   return true;
 }
 
+u32 WC24SendList::GetNumberOfMail() const
+{
+  return Common::swap32(m_data.header.number_of_mail);
+}
+
+u32 WC24SendList::GetEntryId(u32 entry_index) const
+{
+  return Common::swap32(m_data.entries[entry_index].id);
```

Comment:
I haven't traced an exact path for this (probably a corrupted/malicious entry in `next_entry_id` of the on-disk file could do it?) but to me it seems entirely possible for this `entry_index` to access out of bounds. You might want to check for that, and/or sanitize the data on readin in `ReadSendList()`.